pkgs/development/libraries/qt-5/5.9/qtbase/default.nix

```diff
@@ -45,7 +45,7 @@ stdenv.mkDerivation {
       libjpeg libpng libtiff
     ]
 
-    ++ lib.optional mesaSupported mesa
+    ++ lib.optional (mesaSupported && !stdenv.isDarwin) mesa
```

Comment:
This already defaults to false on darwin.

Reply:
On Darwin, the mesaSupported is passed in as true instead of using default value (false). That's the reason why qtdeclarative cannot build. If you look at output of qtbase configuration, the current result shows OpenVG is yes due to mesaSupported value as true.